test/storage/b_plus_tree_insert_test.cpp

```diff
@@ -130,24 +130,24 @@ TEST(BPlusTreeTests, DISABLED_InsertTest3) {
 
   int64_t start_key = 1;
   int64_t current_key = start_key;
-  index_key.SetFromInteger(start_key);
-  for (auto iterator = tree.Begin(index_key); iterator != tree.End(); ++iterator) {
-    auto location = (*iterator).second;
+  for (auto iter = tree.Begin(); iter != tree.End(); ++iter) {
+    auto pair = *iter;
+    auto location = pair.second;
     EXPECT_EQ(location.GetPageId(), 0);
     EXPECT_EQ(location.GetSlotNum(), current_key);
-    ++current_key;
+    current_key = current_key + 1;
   }
 
   EXPECT_EQ(current_key, keys.size() + 1);
 
   start_key = 3;
   current_key = start_key;
   index_key.SetFromInteger(start_key);
-  for (auto iterator = tree.Begin(index_key); iterator != tree.End(); ++iterator) {
+  for (auto iterator = tree.Begin(index_key); !iterator.IsEnd(); ++iterator) {
     auto location = (*iterator).second;
     EXPECT_EQ(location.GetPageId(), 0);
     EXPECT_EQ(location.GetSlotNum(), current_key);
-    ++current_key;
+    current_key = current_key + 1;
```

Comment:
Feel like the previous one is better...

Reply:
They are same and I think the latter one can achieve better synchronization!